### PR TITLE
Custom LogDNA handler and formatter (Guzzle)

### PR DESCRIPTION
**Pull request by ziadoz**
This handler uses Guzzle, which defaults to cURL and fallbacks to streams. I've experience similar but less frequent timeout issues with the default of 5 seconds here.

### CODE REVIEW

File: tests/Formatter/SmartJsonFormatterTest.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {
+        $record = $this->getRecord(Logger::INFO, 'This is a test message', [
+            'exception' => $this->getExceptionWithStackTrace('This is a test exception', 42, null),
+        ]);
+
+        $formatter = new SmartJsonFormatter;
+        $output    = json_decode($formatter->format($record), true);
+
+        // Core fields.
+        $this->assertSame($record['datetime']->getTimestamp(), $output['lines'][0]['timestamp']);
```
Comment (liamkeily) [inline tests/Formatter/SmartJsonFormatterTest.php:22]: Is there a reason we can't just use a fixture file to validate all this json?

File: tests/fixtures/log_lines.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
```
Comment (liamkeily) [inline tests/fixtures/log_lines.json]: This fixture file doesn't appear to be used

**Comment by liamkeily** [comment]
This looks nicely put together. Just for the sake of choosing one i'd vote for the Guzzle version, it sounds like it's more mature and performed slightly better in tests. Either way it sounds like we have the flexibility to experiment with curl settings (timeouts / http2) to figure out what's best for us internally.

**Review by maxakropolis — APPROVED** [review]
Looks good to me. Agree with @liamkeily about the tests, but otherwise 👍 
Guzzle over Symfony for me

File: tests/fixtures/log_lines.json (hunk 2 above)
Comment (ziadoz) [reply tests/fixtures/log_lines.json]: @liamkeily Yeah, it's a hang up from when I forked the Symfony branch. I'll remove it.

File: tests/Formatter/SmartJsonFormatterTest.php (hunk 1 above)
Comment (ziadoz) [reply tests/Formatter/SmartJsonFormatterTest.php:22]: @liamkeily The initial issue I had was timestamps not matching up, but I've fixed the timestamp in the test log record we use and it works, so all the JSON formatting is tested via fixtures.

**Review by weshooper — APPROVED** [review]
Nice job @ziadoz 😃

**Comment by ziadoz** [comment]
@fusionspim/devs I've restored the assertions for the smart JSON formatter here, because the output contains a file path, which differs locally vs on Semaphore.

**Review by weshooper — APPROVED** [review]

**Pull request merged**